Add files via upload

kolomietsbogdan opened the pull request:
Тест о рефакторинге, задача 2. Описание лежит в ридми в мэйне.

kolomietsbogdan committed:
Add files via upload

kolomietsbogdan committed:
Update CheckDateTimeValidity.cpp

Меняю слетевшую кодировку закомментированной строки

Daoroman commented on CheckDateTimeValidity.cpp:
```diff
@@ -0,0 +1,89 @@
+    }
+
+private:
+    static void ValidateYear(int year) {
```
Добавление метода по твоему совету void CheckDataRange(int data, int min, int max, const string error) {
    if (data < min) {
        throw domain_error(error+ " is too small"s);
    }
    if (data > max) {
        throw domain_error(error + " is too big"s );
    }
} 
Существенно сократило код. Думаю этот же метод можно записать и в класс, что так же существенно облегчит класс.

kolomietsbogdan committed:
Update CheckDateTimeValidity.cpp

Сделал шаблонный метод CheckDataRange, который принимает значения data, min, max и error. Код подсократился, да)

kolomietsbogdan commented:
Обновил код CheckDateTimeValidity, теперь всё выглядит компактнее)

Daoroman commented:
Код и правда выглядит куда лучше! Отличная идея.

kolomietsbogdan commented:
Благодарю за ревью!
